api/src/main/java/org/openmrs/util/MemoryAppender.java

```diff
@@ -76,16 +76,14 @@ protected static MemoryAppender createAppender(
 		MemoryAppender appender = null;
 		if (APPENDERS.containsKey(name)) {
 			appender = APPENDERS.get(name).get();
-
-			if (appender != null) {
-				if (appender.bufferSize != theBufferSize) {
-					ThreadSafeCircularFifoQueue<LogEvent> oldBuffer = appender.buffer;
-					appender.buffer = new ThreadSafeCircularFifoQueue<>(theBufferSize);
-					appender.bufferSize = theBufferSize;
+			if (appender != null && appender.bufferSize != theBufferSize) {
+				ThreadSafeCircularFifoQueue<LogEvent> oldBuffer = appender.buffer;
+				appender.buffer = new ThreadSafeCircularFifoQueue<>(theBufferSize);
+				appender.bufferSize = theBufferSize;
+				appender.buffer.addAll(oldBuffer);
 					appender.buffer.addAll(oldBuffer);
 				}
 			}
```

Comment:
![image](https://user-images.githubusercontent.com/9057882/91222041-ea0d5980-e71e-11ea-928e-31708b87bc53.png)

Reply:
only line 84 should be changed here, finished it